src/capwap_ac.erl

```diff
@@ -2626,15 +2626,22 @@ add_location(NameAtom, Map) when is_atom(NameAtom) ->
     add_location(atom_to_binary(NameAtom, utf8), Map);
 add_location(Name, Map) when is_binary(Name) ->
     ?LOG(debug, "Getting location for: ~p", [Name]),
-    AVPName = capwap_config:get(ac, location_avp, undefined),
     case capwap_loc_provider:get_loc(Name, false) of
         {location, LatVal, LongVal} ->
             ?LOG(debug, "Successful location: ~p", [{LatVal, LongVal}]),
-            Map#{AVPName => iolist_to_binary(io_lib:format("Lat:~s;Lon:~s", [LatVal, LongVal]))};
+            Map#{'IM-LI-Location' =>
+		     #{'CAPWAP-GPS-Latitude' => bin_to_float(LatVal),
+		       'CAPWAP-GPS-Longitude' => bin_to_float(LongVal)}};
         {error, Cause} ->
-            ?LOG(error, "Error retrieving location: ~p", [Cause]),
+            ?LOG(warning, "Error retrieving location: ~p", [Cause]),
             Map
     end;
 add_location(Name, Map) ->
     ?LOG(notice, "Name provided is not a binary or an atom, avoiding location retrieval: ~p", [Name]),
     Map.
+
+bin_to_float(BinVal) ->
+    try binary_to_float(BinVal)
+    catch error:badarg ->
+	float(binary_to_integer(BinVal))
```

Comment:
I know that this is not expected here but still - what if the `BinVal` will be a list or float or other type?

Reply:
The type for the loc config providers will expect the result to be `{error, any()} | {location, <<>>, <<>>})`, so a type check will find the issue. There's a bit more type spec'ing to do, but the result is specified to be a binary.